Proteomics/Modifications/ModificationGeneral.cs

```diff
@@ -0,0 +1,218 @@
+                    {
+                        List<string> myValues = new List<string>(this.TaxonomicRange[myKey]);
+                        myValues.Sort();
+                        foreach (string myValue in myValues)
+                        {
+                            sb.AppendLine("TR   " + myKey + "; " + myValue);
+                        }
+                    }
+                }              
+            }
+            if (this.Keywords != null)
+            {
+                if(this.Keywords.Count != 0)
+                {
+                    sb.Append("KW   " + String.Join(" or ", this.Keywords));
+                }                
+            }
+            if (this.NeutralLosses != null)
+            {
+                if(this.NeutralLosses.Count != 0)
+                {
+                    string myLine = "NL   ";
+                    List<DissociationType> myKeys = new List<DissociationType>(this.NeutralLosses.Keys);
+                    myKeys.Sort();
+                    foreach (DissociationType myKey in myKeys)
+                    {
+                        List<double> myValues = new List<double>(this.NeutralLosses[myKey]);
+                        myValues.Sort();
+                        for (int i = 0; i < myValues.Count; i++)
+                        {
+                            myLine = myLine + myKey + ":" + myValues[i];
+                            if (i < myValues.Count)
+                                myLine = myLine + " or ";
+                        }
+                    }
+                    sb.Append(myLine);
+                }             
+            }
+            if (this.DiagnosticIons != null || this.DiagnosticIons.Count != 0)
+            {
+                if(this.DiagnosticIons.Count != 0)
+                {
+                    string myLine = "DI   ";
+                    List<DissociationType> myKeys = new List<DissociationType>(this.DiagnosticIons.Keys);
+                    myKeys.Sort();
+                    foreach (DissociationType myKey in myKeys)
+                    {
+                        List<double> myValues = new List<double>(this.DiagnosticIons[myKey]);
+                        myValues.Sort();
+                        for (int i = 0; i < myValues.Count; i++)
+                        {
+                            myLine = myLine + myKey + ":" + myValues[i];
+                            if (i < myValues.Count)
+                                myLine = myLine + " or ";
+                        }
+                    }
+                    sb.Append(myLine);
+                }            
+            }
+
+            if (this.Keywords != null)
+                if(this.Keywords.Count != 0)
+                    sb.AppendLine("KW   " + String.Join(" or ",  this.Keywords.ToList().OrderBy(b=>b)));
+            return sb.ToString();
+        }
+        public string ModificationErrorsToString() //reports errors in required fields.
```

Comment:
Can you use CodeMaid to clean up the classes you modified, e.g. whitespace and usings?